### PR TITLE
Positioned.fromRelativeRect

### PR DESCRIPTION
**Pull request by Hixie**
Since RelativeRect's whole purpose in life is to make work with Stack
easier, it's silly that you can't directly use it with Positioned.

Also, tests for RelativeRect.
And fixes for the bugs that found...

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by HansMuller** [comment]
@abarth Yes, this is the constructor you asked for: https://github.com/flutter/flutter/pull/8112#discussion_r100925404

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -1668,6 +1668,22 @@ class Positioned extends ParentDataWidget<Stack> {
        bottom = null,
        super(key: key, child: child);
 
+  /// Creates a Positioned object with the values from the given [RelativeRect].
+  ///
+  /// This sets the [left], [top], [right], and [bottom] properties from the
+  /// given [RelativeRect]. The [height] and [width] properties are set to null.
+  Positioned.fromRelativeRect({
+    Key key,
+    Widget child,
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/basic.dart]: Isn't the child @required (likewise below)? Also: if you're willing to be obsessive, the Positioned constructor lists the required child last.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -1668,6 +1668,22 @@ class Positioned extends ParentDataWidget<Stack> {
...
+    RelativeRect rect,
+  }) : left = rect.left,
+       top = rect.top,
+       right = rect.right,
+       bottom = rect.bottom,
+       width = null,
+       height = null,
+       super(key: key, child: child);
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/basic.dart]: You could use rect?.left etc here and then assert rect != null. Maybe that's an infinitesimally better failure mode :-).

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: Fixed.

**Comment by Hixie** [comment]
will land on green

**Pull request merged**